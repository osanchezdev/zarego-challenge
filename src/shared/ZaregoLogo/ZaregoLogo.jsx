import React from 'react';
import { func } from 'prop-types';
import logo from '../../assets/images/logo.png';
import styles from './ZaregoLogo.module.css';

const ZaregoLogo = ({ handleClick }) => (
  <div>
    <img
      className={`${styles.zarego_logo}${
        !handleClick ? styles.default_logo : ''
      }`}
      src={logo}
      alt="Zarego"
      onClick={handleClick}
    />
  </div>
);

ZaregoLogo.propTypes = {
  handleClick: func,
};

ZaregoLogo.defaultProps = {
  handleClick: null,
};

export default React.memo(ZaregoLogo);
