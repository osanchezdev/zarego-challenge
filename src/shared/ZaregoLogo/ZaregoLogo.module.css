.zarego_logo {
  cursor: pointer;
  max-width: 100%;
}
.zarego_logo .default_logo {
  cursor: default;
}
