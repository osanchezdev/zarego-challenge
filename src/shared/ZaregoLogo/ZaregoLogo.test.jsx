import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ZaregoLogo from './ZaregoLogo';

describe('ZaregoLogo component', () => {
  it('should render component', () => {
    const tree = renderer.create(<ZaregoLogo />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should execute handleClick', () => {
    const handleClick = jest.fn();
    render(<ZaregoLogo handleClick={handleClick} />);
    const logo = screen.getByAltText(/zarego/i);

    fireEvent.click(logo);

    expect(handleClick).toBeCalledTimes(1);
  });
});
