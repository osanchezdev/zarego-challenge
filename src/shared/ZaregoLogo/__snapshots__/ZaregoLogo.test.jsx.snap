// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZaregoLogo component should render component 1`] = `
<div>
  <img
    alt="Zarego"
    className="zarego_logodefault_logo"
    onClick={null}
    src="logo.png"
  />
</div>
`;
